Week4/Lecture8/Assignment3/lecture8_assignment3/src/main/java/com/example/lecture8_assignment3/service/EmployeeService2.java

```diff
@@ -0,0 +1,40 @@
+package com.example.lecture8_assignment3.service;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.example.lecture8_assignment3.model.Employee;
+import com.example.lecture8_assignment3.repository.EmployeeRepository2;
+
+@Service
+public class EmployeeService2 {
+
+    @Autowired
+    private EmployeeRepository2 employeeRepository2;
+
+    public List<Employee> getAllEmployees() {
+        return employeeRepository2.findAll();
+    }
+
+    public Employee getEmployeeById(int id) {
+        return employeeRepository2.findById(id);
+    }
+
+    @Transactional("transactionManager2")
```

Comment:
Question: What is case @Transactional does not work and explain.

Reply:
some scenarios where **@transactional** does not work:
- annotating non-public methods will result in no transactional behavior
- When a method annotated with @Transactional calls another method within the same class directly, the transactional behavior will not be applied to the called method.
- If transaction management is not properly enabled, the @Transactional annotation will not work
- If you specify a transaction manager that does not match the data source being used, transactions will not work as expected.

Reply:
It's very good points